src/crush/CrushWrapper.cc

```diff
@@ -2347,7 +2347,12 @@ int CrushWrapper::_choose_type_stack(
       underfull_buckets[j].insert(item);
     }
   }
-  ldout(cct, 20) << __func__ << " underfull_buckets " << underfull_buckets << dendl;
+  ldout(cct, 20) << __func__ << " underfull_buckets " << "[";
+  for (auto p = underfull_buckets.begin(); p != underfull_buckets.end(); ++p) {
+    if (p != underfull_buckets.begin()) *_dout << ",";
+    *_dout << *p;
+  }
+  *_dout << "]" << dendl;
 
```

Comment:
From the commit description:

If you are talking about the following << operator function in include/types.h, then I don't think it will work for the below statement, just by looking at the vector<A, Alloc>. In this case, underfull_buckets doesn't match that template. I think we can overload << for vector<set<int>>, if such instances happen many times, which is the best solution. So please check that.

> inline ostream& operator<<(ostream& out, const vector<A,Alloc>& v) {
> ldout(cct, 20) << __func__ << " underfull_buckets " << underfull_buckets << dendl;